src/devices/Shared/Sensirion.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Iot.Device
+{
+    /// <summary>
+    /// Common code for Sensirion devices.
+    /// </summary>
+    internal static class Sensirion
+    {
+        /// <returns>If the CRC8 matched the data, a <see cref="ushort"/> of data. Otherwise, <see langword="null"/>.</returns>
+        public static ushort? ReadUInt16BigEndianAndCRC8(ReadOnlySpan<byte> bytes)
+        {
+            Debug.Assert(bytes.Length >= 3, $"{nameof(bytes)} must contain at least 3 bytes.");
+
+            _ = bytes[2];
```

Comment:
mm not sure I understand why you have this line.

Reply:
Micro-optimization. It forces a single bounds check on the span. Otherwise, the in-order access to `bytes[0]`, `bytes[1]`, `bytes[2]` will result in three bounds checks.